cmake/FindmbedTLS.cmake

```diff
@@ -0,0 +1,82 @@
+# - Try to find mbedTLS
+# Once done this will define
+#
+#  MBEDTLS_ROOT_DIR - Set this variable to the root installation of mbedTLS
+#
+# Read-Only variables
+#  MBEDTLS_FOUND - system has mbedTLS
+#  MBEDTLS_INCLUDE_DIR - the mbedTLS include directory
+#  MBEDTLS_LIBRARIES - Link these to use mbedTLS
+#
+
+FIND_PATH(MBEDTLS_ROOT_DIR NAMES include/mbedtls/version.h)
+
+IF(MBEDTLS_INCLUDE_DIR AND MBEDTLS_LIBRARIES)
+    # Already in cache, be silent
+    SET(MBEDTLS_FIND_QUIETLY TRUE)
+ENDIF()
+FIND_PATH(MBEDTLS_INCLUDE_DIR
+    NAMES mbedtls/ssl.h
+    PATHS
+        ${MBEDTLS_ROOT_DIR}/include
+)
+FIND_LIBRARY(mbedtls_lib
+    NAMES mbedtls libmbedtls libmbedx509
+    PATHS
+        ${MBEDTLS_ROOT_DIR}/library
+        ${MBEDTLS_ROOT_DIR}/build/library
```

Comment:
these should probably refer to an install tree, not a build tree (which certainly isn't guaranteed to be called "build")